selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/factory/TestFactory.java

```diff
@@ -48,10 +50,10 @@ public TestFactory(
 
   /** Adds authentication token into the browser and opens factory url. */
   public void authenticateAndOpen(SeleniumWebDriver driver) throws Exception {
-    driver.get(dashboardUrl.get().toString());
-
+    String dashboardUrlPath = dashboardUrl.get().toString();
+    driver.get(dashboardUrlPath);
     entrance.login(owner);
-
+    waitUntilTextPresentInUrl(driver, dashboardUrlPath + "#/");
```

Comment:
"#/" looks like a kind of rudiment

Reply:
This is a part which added when dashboard start loading, and exactly this we need check. After that webdriver can open another url without mistakes.

Reply:
It's a kind of workaround. Can we wait on presence of some element on page?

Reply:
We already have login page. IMHO it's more reasonable to check if page has already been loaded inside the **entrance.login()** method.